src/form/DecompositionTable/DecompositionTable.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import { TableHeader, TableBody, TableRow, TableCell, Switch, ContextMenuItem } from '@amsterdam/asc-ui';
+import { Ellipsis } from '@amsterdam/asc-assets';
+import { StyledDecompositionTable, StyledContextMenu } from './DecompositionTableStyles'
+
+
+export type Props = {
+	rows: any[];
+};
+
+const DecompositionTable: React.FC<Props> = ({ rows }: Props) => {
+
+	return (
+		<StyledDecompositionTable>
+			<TableHeader>
+				<TableRow>
+					<TableCell as="th">Code</TableCell>
+					<TableCell as="th">Deel</TableCell>
+					<TableCell as="th">Materiaal</TableCell>
+					<TableCell as="th">Hoeveelheid</TableCell>
+					<TableCell as="th">Eenheid</TableCell>
+					<TableCell as="th">Locatie</TableCell>
+					<TableCell as="th">Jaar</TableCell>
+					<TableCell as="th">Relevant</TableCell>
+					<TableCell as="th">Acties</TableCell>
+				</TableRow>
+			</TableHeader>
+			<TableBody>
+				{rows.map((row) => (
+					<TableRow
+						key={row.id}
+						className={`row ${row.elementId && !row.unitId ? 'row__unit' : ''} ${row.unitId ? 'row__manifestation' : ''
```

Comment:
This is not really the styled-components way of doing this.
If TableRow is a styled component it can use props. You can pass it the entire row object and let it conditionally apply the appropriate styles. By setting class names without importing them from your styles file, the class names are not encapsulated to the component and are left vulnerable for cross component contamination.

Reply:
https://styled-components.com/docs/faqs#how-can-i-override-styles-with-higher-specificity